bigquery/google/cloud/bigquery/external_config.py

```diff
@@ -0,0 +1,485 @@
+            config[self.options._RESOURCE_NAME] = self.options.to_api_repr()
+        return config
+
+    @classmethod
+    def from_api_repr(cls, resource):
+        """Factory: construct a CSVOptions given its API representation
+
+        :type resource: dict
+        :param resource:
+            An extract job configuration in the same representation as is
+            returned from the API.
+
+        :rtype: :class:`google.cloud.bigquery.external_config.CSVOptions`
+        :returns: Configuration parsed from ``resource``.
+        """
+        config = cls()
+        schema = resource.pop('schema', None)
+        if schema:
+            config.schema = _parse_schema_resource(schema)
+        for optcls in (BigtableOptions, CSVOptions, GoogleSheetsOptions):
+            opts = resource.pop(optcls._RESOURCE_NAME, None)
+            if opts is not None:
+                config.options = optcls.from_api_repr(opts)
+                break
+        config._properties = copy.deepcopy(resource)
+        return config
+
+
+class BigtableColumn(object):
+    """Options for a Bigtable column."""
+
+    def __init__(self):
+        self._properties = {}
+
+    encoding = _TypedApiResourceProperty(
+        'encoding', 'encoding', six.string_types)
+    """See
+    https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.query.tableDefinitions.(key).bigtableOptions.columnFamilies.columns.encoding
+    https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#externalDataConfiguration.bigtableOptions.columnFamilies.columns.encoding
+    """
+
+    field_name = _TypedApiResourceProperty(
+        'field_name', 'fieldName', six.string_types)
+    """See
+    https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.query.tableDefinitions.(key).bigtableOptions.columnFamilies.columns.field_name
+    https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#externalDataConfiguration.bigtableOptions.columnFamilies.columns.field_name
+    """
+
+    only_read_latest = _TypedApiResourceProperty(
+        'only_read_latest', 'onlyReadLatest', bool)
+    """See
+    https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.query.tableDefinitions.(key).bigtableOptions.columnFamilies.columns.only_read_latest
+    https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#externalDataConfiguration.bigtableOptions.columnFamilies.columns.only_read_latest
+    """
+
+    qualifier_encoded = _TypedApiResourceProperty(
+        'qualifier_encoded', 'qualifierEncoded', six.binary_type)
+    """See
+    https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.query.tableDefinitions.(key).bigtableOptions.columnFamilies.columns.qualifier_encoded
+    https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#externalDataConfiguration.bigtableOptions.columnFamilies.columns.qualifier_encoded
+    """
+
+    qualifier_string = _TypedApiResourceProperty(
```

Comment:
We could potentially do like you do for options and take care of populating the right field depending on if we are given a binary value or a unicode value.

Reply:
Yes, but it's very subtle because of the different binary/text types in Python 2 vs. 3. For example, if someone passes in `str`, that will be binary in Python 2 but text in Python 3.

Probably safer to have less magic.

Reply:
We do a little bit of this kind of magic in the DB-API,

https://github.com/GoogleCloudPlatform/google-cloud-python/blob/master/bigquery/google/cloud/bigquery/dbapi/_helpers.py#L49

But, I'm okay erring on the side of less magic.